erpnext/asset/doctype/asset/asset.js

```diff
@@ -0,0 +1,304 @@
+				}
+			};
+		});
+	},
+
+	refresh: function(frm) {
+		frappe.ui.form.trigger("Asset", "is_existing_asset");
+		frm.toggle_display("next_depreciation_date", frm.doc.docstatus < 1);
+		frm.events.make_schedules_editable(frm);
+
+		if (frm.doc.docstatus==1) {
+			if (in_list(["Submitted", "Partially Depreciated", "Fully Depreciated"], frm.doc.status)) {
+				frm.add_custom_button("Transfer Asset", function() {
+					erpnext.asset.transfer_asset(frm);
+				});
+
+				frm.add_custom_button("Scrap Asset", function() {
+					erpnext.asset.scrap_asset(frm);
+				});
+
+				frm.add_custom_button("Sell Asset", function() {
+					erpnext.asset.make_sales_invoice(frm);
+				});
+
+			} else if (frm.doc.status=='Scrapped') {
+				frm.add_custom_button("Restore Asset", function() {
+					erpnext.asset.restore_asset(frm);
+				});
+			}
+			if (frm.doc.status=='Submitted' && !frm.doc.is_existing_asset && !frm.doc.purchase_invoice) {
+				frm.add_custom_button(__("Purchase Invoice"), function() {
+					frm.trigger("make_purchase_invoice");
+				}, __("Make"));
+			}
+			if (frm.doc.maintenance_required && !frm.doc.maintenance_schedule) {
+				frm.add_custom_button(__("Asset Maintenance"), function() {
+					frm.trigger("create_asset_maintenance");
+				}, __("Make"));
+			}
+			frm.page.set_inner_btn_group_as_primary(__("Make"));
+			frm.trigger("setup_chart");
+		}
+	},
+
+	setup_chart: function(frm) {
+		var x_intervals = [frm.doc.purchase_date];
+		var asset_values = [frm.doc.gross_purchase_amount];
+		var last_depreciation_date = frm.doc.purchase_date;
+
+		if(frm.doc.opening_accumulated_depreciation) {
+			last_depreciation_date = frappe.datetime.add_months(frm.doc.next_depreciation_date,
+				-1*frm.doc.frequency_of_depreciation);
+
+			x_intervals.push(last_depreciation_date);
+			asset_values.push(flt(frm.doc.gross_purchase_amount) -
+				flt(frm.doc.opening_accumulated_depreciation));
+		}
+
+		$.each(frm.doc.schedules || [], function(i, v) {
+			x_intervals.push(v.schedule_date);
+			var asset_value = flt(frm.doc.gross_purchase_amount) - flt(v.accumulated_depreciation_amount);
+			if(v.journal_entry) {
+				last_depreciation_date = v.schedule_date;
+				asset_values.push(asset_value);
+			} else {
+				if (in_list(["Scrapped", "Sold"], frm.doc.status)) {
+					asset_values.push(null);
+				} else {
+					asset_values.push(asset_value)
+				}
+			}
+		});
+
+		if(in_list(["Scrapped", "Sold"], frm.doc.status)) {
+			x_intervals.push(frm.doc.disposal_date);
+			asset_values.push(0);
+			last_depreciation_date = frm.doc.disposal_date;
+		}
+
+		frm.dashboard.render_graph({
+			title: "Asset Value",
+			data: {
+				labels: x_intervals,
+				datasets: [{
+					color: 'green',
+					values: asset_values,
+					formatted: asset_values.map(d => d.toFixed(2))
+				}]
+			},
+			type: 'line'
+		});
+	},
+
+
+	item_code: function(frm) {
+		if(frm.doc.item_code) {
+			frappe.call({
+				method: "erpnext.asset.doctype.asset.asset.get_item_details",
```

Comment:
please check if `erpnext.asset.doctype.asset.asset.get_item_details` can be replaced from `frappe.client.get`. With the proper filtering.